### PR TITLE
sync neo

### LINKED ISSUE #742: neo changes need to be synchronized
In https://github.com/neo-project/neo/pull/2314.

### PR DESCRIPTION
**Pull request by bettybao1209**
Close #742

### CODE REVIEW

File: neo-cli/CLI/MainService.Wallet.cs [hunk 1]
```diff
@@ -226,7 +226,7 @@ private void OnImportMultisigAddress(ushort m, ECPoint[] publicKeys)
             if (CurrentWallet is NEP6Wallet wallet)
                 wallet.Save();
 
-            Console.WriteLine("Multisig. Addr.: " + multiSignContract.Address);
+            Console.WriteLine("Multisig. Addr.: " + Contract.GetBFTAddress(publicKeys));
```
Comment (shargon) [inline neo-cli/CLI/MainService.Wallet.cs:229]: is not the same, where is `m`?

Comment (bettybao1209) [reply neo-cli/CLI/MainService.Wallet.cs]: Fixed.

**Review by shargon — APPROVED** [review]

**Comment by erikzhang** [comment]
https://github.com/neo-project/neo-node/blob/83fe4d154c4a03cb09c9e3ad32d08305b4c51c2a/neo-cli/Settings.cs#L59

Here we shouldn't use `ProtocolSettings.Default` anymore.

**Comment by bettybao1209** [comment]
wait for https://github.com/neo-project/neo/pull/2336

File: neo-cli/CLI/MainService.cs [hunk 2]
```diff
@@ -36,6 +36,8 @@ public partial class MainService : ConsoleServiceBase, IWalletProvider
         public const long TestModeGas = 20_00000000;
 
         private Wallet currentWallet;
+
+        internal static ProtocolSettings ProtocolSettings;
```
Comment (erikzhang) [inline neo-cli/CLI/MainService.cs:40]: I need to support multiple `NeoSystem` in cli, please don't use static `ProtocolSettings` or `NeoSystem`.

Comment (bettybao1209) [reply neo-cli/CLI/MainService.cs]: Fixed.

**Review by shargon — DISMISSED** [review]

File: neo-cli/CLI/Logger.cs [hunk 3]
```diff
@@ -14,12 +14,18 @@ internal class Logger : Plugin, ILogPlugin
         private static readonly ConsoleColorSet WarningColor = new ConsoleColorSet(ConsoleColor.Yellow);
         private static readonly ConsoleColorSet ErrorColor = new ConsoleColorSet(ConsoleColor.Red);
         private static readonly ConsoleColorSet FatalColor = new ConsoleColorSet(ConsoleColor.Red);
+        private readonly uint magic;
```
Comment (erikzhang) [inline neo-cli/CLI/Logger.cs:17]: I think `Logger` should be network-independent. Because every network may have logs.

**Comment by cloud8little** [comment]
SendTransaction/OnInvokeResult method, should add protocolsettings for parameters.

https://github.com/neo-project/neo-node/blob/83fe4d154c4a03cb09c9e3ad32d08305b4c51c2a/neo-cli/CLI/MainService.cs#L484-L493

https://github.com/neo-project/neo-node/blob/83fe4d154c4a03cb09c9e3ad32d08305b4c51c2a/neo-cli/CLI/MainService.cs#L560-L568

File: neo-cli/CLI/Logger.cs (hunk 3 above)
Comment (bettybao1209) [reply neo-cli/CLI/Logger.cs:17]: Should we add `magic` as a parameter in the `Log` method like this: `ILogPlugin.Log(uint magic, string source, LogLevel level, object message)`

**Comment by ProDog** [comment]
close https://github.com/neo-project/neo-node/issues/742.

File: neo-cli/CLI/Logger.cs (hunk 3 above)
Comment (erikzhang) [reply neo-cli/CLI/Logger.cs:17]: I think it's impossible because Akka logs can't pass these parameters.

**Comment by bettybao1209** [comment]
> SendTransaction/OnInvokeResult method, should add protocolsettings for parameters.

Fixed.

File: neo-cli/CLI/Logger.cs (hunk 3 above)
Comment (shargon) [reply neo-cli/CLI/Logger.cs:17]: Use base path instead of magic?

Comment (erikzhang) [reply neo-cli/CLI/Logger.cs:17]: We can just remove it.

Comment (bettybao1209) [reply neo-cli/CLI/Logger.cs]: Fixed.

**Review by shargon — DISMISSED** [review]

File: neo-cli/CLI/MainService.Node.cs [hunk 4]
```diff
@@ -65,14 +65,13 @@ private void OnShowStateCommand()
             Task task = Task.Run(async () =>
             {
                 int maxLines = 0;
-
+                LocalNode localNode = await NeoSystem.LocalNode.Ask<LocalNode>(new LocalNode.GetInstance());
```
Comment (erikzhang) [inline neo-cli/CLI/MainService.Node.cs:68]: Get `localNode` on start?

Comment (bettybao1209) [reply neo-cli/CLI/MainService.Node.cs]: Done.

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**